### PR TITLE
Rename TxBody and TxBodyRaw fields and patterns without underscores

### PR DESCRIPTION
**Pull request by aniketd**
Partially addresses #3061 
- [x] Update CHANGELOG

### CODE REVIEW

File: eras/shelley-ma/impl/src/Cardano/Ledger/Mary/Value.hs [hunk 1]
```diff
@@ -367,14 +367,14 @@ decodeIntegerBounded64 = do
     TypeUInt64 -> pure ()
     TypeNInt -> pure ()
     TypeNInt64 -> pure ()
-    _ -> fail "expected major type 0 or 1 when decoding mint field"
+    _ -> fail "expected major type 0 or 1 when decoding smatbrMint field"
```
Comment (Soupstraw) [inline eras/shelley-ma/impl/src/Cardano/Ledger/Mary/Value.hs]: I think it would be more readable when we left it to `mint` here

Comment (aniketd) [reply eras/shelley-ma/impl/src/Cardano/Ledger/Mary/Value.hs]: Oops! Yes!

File: eras/shelley/impl/src/Cardano/Ledger/Shelley/TxBody.hs [hunk 2]
```diff
@@ -237,29 +237,29 @@ txSparse (TxBodyRaw input output cert wdrl fee ttl update hash) =
...
-      _ttlX = SlotNo 0,
-      _certsX = StrictSeq.empty,
-      _wdrlsX = Wdrl Map.empty,
-      _txUpdateX = SNothing,
-      _mdHashX = SNothing
+  ShelleyTxBodyRaw
+    { stbrInputs = Set.empty,
+      stbrOutputs = StrictSeq.empty,
+      stbrTxFee = Coin 0,
+      stbrTtl = SlotNo 0,
```
Comment (Soupstraw) [inline eras/shelley/impl/src/Cardano/Ledger/Shelley/TxBody.hs]: I think `stbrTTL` would be better, since it's an abbreviation (time-to-live)

File: eras/shelley/test-suite/src/Test/Cardano/Ledger/Shelley/Address/Bootstrap.hs [hunk 3]
```diff
@@ -225,14 +225,14 @@ coinsToBob = Coin 1000
...
-      _txUpdate = SNothing,
-      _mdHash = SNothing
+    { sbtInputs = Set.fromList [TxIn genesisId minBound],
+      sbtOutputs = StrictSeq.fromList [ShelleyTxOut bobAddr coinsToBob, ShelleyTxOut aliceAddr change],
+      sbtCerts = StrictSeq.fromList mempty,
+      sbtWdrls = Wdrl Map.empty,
+      sbtTxfee = fee,
+      sbtTtl = SlotNo 10,
+      sbtTxUpdate = SNothing,
+      sbtMdHash = SNothing
```
Comment (Soupstraw) [inline eras/shelley/test-suite/src/Test/Cardano/Ledger/Shelley/Address/Bootstrap.hs]: `sbtMDHash` would be better here

File: eras/shelley-ma/impl/src/Cardano/Ledger/ShelleyMA/Rules/Utxo.hs [hunk 4]
```diff
@@ -180,7 +180,7 @@ utxoTransition = do
   -- runValidationTransMaybe fromShelleyFailure $ Shelley.validateInputSetEmptyUTxO txb
   runTest $ Shelley.validateInputSetEmptyUTxO txb
 
-  {- minfee pp tx ≤ txfee txb -}
+  {- minfee pp tx ≤ smatbrTxFee txb -}
```
Comment (lehins) [inline eras/shelley-ma/impl/src/Cardano/Ledger/ShelleyMA/Rules/Utxo.hs]: We should revert the changes in all comments, since those are snippets from the spec
```suggestion
  {- minfee pp tx ≤ txfee txb -}
```

File: eras/shelley-ma/impl/src/Cardano/Ledger/ShelleyMA/Rules/Utxo.hs [hunk 5]
```diff
@@ -201,8 +201,8 @@ utxoTransition = do
   ppup' <-
     trans @(EraRule "PPUP" era) $ TRC (PPUPEnv slot pp genDelegs, ppup, txup tx)
 
-  {- adaPolicy ∉ supp mint tx
-     above check not needed because mint field of type MultiAsset cannot contain ada -}
+  {- adaPolicy ∉ supp smatbrMint tx
+     above check not needed because smatbrMint field of type MultiAsset cannot contain ada -}
```
Comment (lehins) [inline eras/shelley-ma/impl/src/Cardano/Ledger/ShelleyMA/Rules/Utxo.hs]: Same here

File: eras/alonzo/impl/src/Cardano/Ledger/Alonzo/TxBody.hs [hunk 6]
```diff
@@ -122,40 +122,40 @@ import Prelude hiding (lookup)
...
-    _outputs :: !(StrictSeq (Core.TxOut era)),
-    _certs :: !(StrictSeq (DCert (EraCrypto era))),
-    _wdrls :: !(Wdrl (EraCrypto era)),
-    _txfee :: !Coin,
-    _vldt :: !ValidityInterval,
-    _update :: !(StrictMaybe (Update era)),
-    _reqSignerHashes :: Set (KeyHash 'Witness (EraCrypto era)),
-    _mint :: !(MultiAsset (EraCrypto era)),
+data AlonzoTxBodyRaw era = AlonzoTxBodyRaw
+  { atbrInputs :: !(Set (TxIn (EraCrypto era))),
```
Comment (teodanciu) [inline eras/alonzo/impl/src/Cardano/Ledger/Alonzo/TxBody.hs:126]: I find these names a bit difficult to process, because of how alien these abbreviations seem (`atbr`, `btbr`, `smat`) - only because they haven't been used, yet so the don't mean anything at the moment. 

On the other hand, I see there's not too many references to them, so maybe it's not so important. 
Just wondering, have you considered using the era name as a prefix? Would it be too long or maybe conflicting with something else? 
I mean 
```
alonzoInputs :: 
alonzoCollateral :: .. 
....
babbageSpendInputs  :: ... 

etc
```

Comment (aniketd) [reply eras/alonzo/impl/src/Cardano/Ledger/Alonzo/TxBody.hs:126]: I completely agree with you on this @teodanciu. The abbreviations are rather unpronounciable. I don't think they are likely to conflict with anything else, though the names might get "too long" - this is a subjective presumption. I can change this to using full era names if you think it will be helpful.

Comment (teodanciu) [reply eras/alonzo/impl/src/Cardano/Ledger/Alonzo/TxBody.hs:126]: Maybe let's see what others think about it before changing, maybe someone has a better idea altogether!

Comment (Soupstraw) [reply eras/alonzo/impl/src/Cardano/Ledger/Alonzo/TxBody.hs:126]: I use a similar style as Aniket, so I'm used to these kinds of names

Comment (lehins) [reply eras/alonzo/impl/src/Cardano/Ledger/Alonzo/TxBody.hs:126]: Same here, I was able to decipher `atbr` for `alonzo tx body raw` just fine. This is a common convention in Haskell, in order to avoid really long names.

The important part is the consistency. This gives us a predictable way of uniquely identifying the type the field works with. The naming convention of using scientists' names in alphabetical order helps us avoid those clashes as well if we use just the first latter (eg, `a` for `Alonzo`, `b` for `Babbage`)


Also with respect to all of the parameterized families, `Tx` , `TxBody`, etc. we should use lenses anyways, so these record field names are mostly an internal implementation detail.

File: eras/shelley/test-suite/src/Test/Cardano/Ledger/Shelley/Generator/ShelleyEraGen.hs [hunk 7]
```diff
@@ -76,11 +76,11 @@ instance
   genEraTxBody _ge _utxo = genTxBody
   genEraAuxiliaryData = genMetadata
 
-  updateEraTxBody _utxo _pp _wits body' fee ins out =
+  updateEraTxBody _utxo _pp _trWits body' fee ins out =
```
Comment (lehins) [inline eras/shelley/test-suite/src/Test/Cardano/Ledger/Shelley/Generator/ShelleyEraGen.hs]: Wrongful search and replacing function arguments. There are many occurrences of this, could you please revert them back
```suggestion
  updateEraTxBody _utxo _pp _wits body' fee ins out =
```

**Review by lehins — CHANGES_REQUESTED** [review]
There are a few accidental renames that weren't suppose to happen,other than that it looks pretty good

File: eras/shelley/test-suite/src/Test/Cardano/Ledger/Shelley/Generator/ShelleyEraGen.hs (hunk 7 above)
Comment (aniketd) [reply eras/shelley/test-suite/src/Test/Cardano/Ledger/Shelley/Generator/ShelleyEraGen.hs]: Noted @lehins :memo: :pray: . Will get them reverted.

File: eras/shelley-ma/impl/src/Cardano/Ledger/ShelleyMA/Rules/Utxo.hs (hunk 4 above)
Comment (aniketd) [reply eras/shelley-ma/impl/src/Cardano/Ledger/ShelleyMA/Rules/Utxo.hs]: Noted! Fixed!

File: eras/shelley-ma/impl/src/Cardano/Ledger/ShelleyMA/Rules/Utxo.hs (hunk 5 above)
Comment (aniketd) [reply eras/shelley-ma/impl/src/Cardano/Ledger/ShelleyMA/Rules/Utxo.hs]: Noted! Fixed!

File: eras/shelley-ma/impl/src/Cardano/Ledger/ShelleyMA/TxBody.hs [hunk 8]
```diff
@@ -287,19 +287,19 @@ pattern TxBody' ::
   StrictMaybe (AuxiliaryDataHash (EraCrypto era)) ->
   MultiAsset (EraCrypto era) ->
   MATxBody era
-pattern TxBody' {inputs', outputs', certs', wdrls', txfee', vldt', update', adHash', mint'} <-
+pattern TxBody' {smatbrInputs', smatbrOutputs', smatbrCerts', smatbrWdrls', smatbrTxFee', smatbrValidityInterval', smatbrTxUpdate', smatbrAuxDataHash', smatbrMint'} <-
```
Comment (lehins) [inline eras/shelley-ma/impl/src/Cardano/Ledger/ShelleyMA/TxBody.hs]: These actually don't need the `r` at the end, since they don't operate on the `Raw` type, but on the . Also removing the ugly tick is a good idea:
```suggestion
pattern TxBody' {matbInputs, matbOutputs, matbCerts, matbWdrls, matbTxFee, matbValidityInterval, matrTxUpdate, matbAuxDataHash, matbMint} <-
```

File: eras/shelley-ma/impl/src/Cardano/Ledger/ShelleyMA/TxBody.hs [hunk 9]
```diff
@@ -104,63 +104,63 @@ import NoThunks.Class (NoThunks (..))
...
-  { inputs :: !(Set (TxIn (EraCrypto era))),
-    outputs :: !(StrictSeq (ShelleyTxOut era)),
-    certs :: !(StrictSeq (DCert (EraCrypto era))),
-    wdrls :: !(Wdrl (EraCrypto era)),
-    txfee :: !Coin,
-    vldt :: !ValidityInterval, -- imported from Timelocks
-    update :: !(StrictMaybe (Update era)),
-    adHash :: !(StrictMaybe (AuxiliaryDataHash (EraCrypto era))),
-    mint :: !(MultiAsset (EraCrypto era))
+data ShelleyMATxBodyRaw era = ShelleyMATxBodyRaw
```
Comment (lehins) [inline eras/shelley-ma/impl/src/Cardano/Ledger/ShelleyMA/TxBody.hs]: For consistency it should be just `MATxBodyRaw`  (in order to match `MATxBody`), which also means that prefix  will need to be switched from `smatbr` -> `matbr`
```suggestion
data MATxBodyRaw era = MATxBodyRaw
```

"Shelley" in the naming of the package is an artifact, so prefixing types with `Shelley` doesn't make much sense (especially since we are planning on splitting into two separate packages `Mary` and `Allegra`

**Review by lehins — CHANGES_REQUESTED** [review]
This is looking very good. I noticed some more stuff that can be improved upon the second review. It would also be nice to make pattern synonym record names for `[Era]TxBody` constant for all eras as well. For example `AlonzoTxBody` is defined as:
```haskell
pattern AlonzoTxBody
  { inputs,
    collateral,
    outputs,
    txcerts,
    txwdrls,
    txfee,
    txvldt,
    txUpdates,
    reqSignerHashes,
    mint,
    scriptIntegrityHash,
    adHash,
    txnetworkid
  }
```

But it should be something like this:
```haskell
pattern AlonzoTxBody
  { atbInputs,
    atbCollateral,
    atbOutputs,
    atbCerts,
    atbWdrls,
    atbTxFee,
    atbValidityInterval,
    atbUpdate,
    atbReqSignerHashes,
    atbMint,
    atbScriptIntegrityHash,
    atbAuxDataHash,
    atbTxNetworkId
  }
```

We want to make sure that there are not a dozen different names for a similar concept (eg `txUpdates` vs `atbrTxUpdate` or `_txUpdate` prior to this PR - note that one is plural another one is not) 

I guess the goal is to have no ambiguity between era and the least amount of discrepancy between record fields for the raw type, the actual newtype wrapper around it as well as the lens names for the corresponding fields.

File: eras/shelley-ma/impl/src/Cardano/Ledger/ShelleyMA/TxBody.hs (hunk 8 above)
Comment (aniketd) [reply eras/shelley-ma/impl/src/Cardano/Ledger/ShelleyMA/TxBody.hs]: Thanks for the suggestion :+1: I also renamed `TxBody'` to `MATxBody_`

File: eras/shelley-ma/impl/src/Cardano/Ledger/ShelleyMA/TxBody.hs (hunk 9 above)
Comment (aniketd) [reply eras/shelley-ma/impl/src/Cardano/Ledger/ShelleyMA/TxBody.hs]: Totally Agree!

File: eras/shelley-ma/impl/src/Cardano/Ledger/ShelleyMA/TxBody.hs [hunk 10]
```diff
@@ -21,20 +21,29 @@ module Cardano.Ledger.ShelleyMA.TxBody
...
+        matbWdrls,
+        matbAuxDataHash_,
+        matbCerts_,
+        matbInputs_,
+        matbMint_,
+        matbOutputs_,
+        matbTxFee_,
+        matbUpdate_,
+        matbValidityInterval_,
+        matbWdrls_
```
Comment (aniketd) [inline eras/shelley-ma/impl/src/Cardano/Ledger/ShelleyMA/TxBody.hs]: Here the set of exports with trailing `_` have been added like the set of exports for identifiers without trailing `_`, owing to the two separate pattern synonyms existing for `MATxBody`

File: eras/alonzo/impl/src/Cardano/Ledger/Alonzo/TxBody.hs [hunk 11]
```diff
@@ -26,19 +26,19 @@ module Cardano.Ledger.Alonzo.TxBody
...
+        atbCerts,
+        atbWdrls,
+        atbTxFee,
+        atbValidityInterval,
+        atbUpdate,
+        atbReqSignerHashes,
+        atbMint,
+        atbScriptIntegrityHash,
+        atbAuxDataHash,
+        atbTxNetworkId
```
Comment (lehins) [inline eras/alonzo/impl/src/Cardano/Ledger/Alonzo/TxBody.hs:41]: Adding these renames to the changelog would probably be a good idea.

File: eras/alonzo/impl/src/Cardano/Ledger/Alonzo/TxBody.hs [hunk 12]
```diff
@@ -300,79 +300,53 @@ pattern AlonzoTxBody ::
...
-            vldtX
-            updateX
-            reqSignerHashesX
-            mintX
-            scriptIntegrityHashX
-            adHashX
-            txnetworkidX
+    AlonzoTxBody inputs collateral outputs certs wdrls txfee validityInterval update reqSignerHashes mint scriptIntegrityHash auxDataHash txnetworkid =
+      mkAlonzoTxBody $
+        AlonzoTxBodyRaw inputs collateral outputs certs wdrls txfee validityInterval update reqSignerHashes mint scriptIntegrityHash auxDataHash txnetworkid
```
Comment (lehins) [inline eras/alonzo/impl/src/Cardano/Ledger/Alonzo/TxBody.hs]: Sorry for asking you to do all this mundane stuff, but could you switch this to use record syntax, it would be much cleaner and follow the pattern above. Something along the lines:
```
AlonzoTxBody {atbInputs, ...} =
  mkAlonzoTxBodyRaw $
    AlonzoTxBodyRaw {
       atbrInputs = atbrInputs,
       ....
       }
````

File: eras/babbage/impl/src/Cardano/Ledger/Babbage/TxBody.hs [hunk 13]
```diff
@@ -37,22 +37,22 @@ module Cardano.Ledger.Babbage.TxBody
     babbageMinUTxOValue,
     BabbageTxBody
       ( BabbageTxBody,
-        inputs,
```
Comment (lehins) [inline eras/babbage/impl/src/Cardano/Ledger/Babbage/TxBody.hs:40]: Same with these ones in the changelog

File: eras/babbage/impl/src/Cardano/Ledger/Babbage/TxBody.hs [hunk 14]
```diff
@@ -533,85 +533,53 @@ pattern BabbageTxBody ::
...
-            vldtX
-            updateX
-            reqSignerHashesX
-            mintX
-            scriptIntegrityHashX
-            adHashX
-            txnetworkidX
+    BabbageTxBody inputs collateral referenceInputs outputs collateralReturn totalCollateral certs wdrls txfee vldt update reqSignerHashes mint scriptIntegrityHash adHash txnetworkid =
+      mkBabbageTxBodyFromRaw $
+        BabbageTxBodyRaw inputs collateral referenceInputs outputs collateralReturn totalCollateral certs wdrls txfee vldt update reqSignerHashes mint scriptIntegrityHash adHash txnetworkid
```
Comment (lehins) [inline eras/babbage/impl/src/Cardano/Ledger/Babbage/TxBody.hs]: Same here with respect to record style syntax

File: eras/shelley-ma/impl/src/Cardano/Ledger/ShelleyMA/TxBody.hs (hunk 10 above)
Comment (lehins) [reply eras/shelley-ma/impl/src/Cardano/Ledger/ShelleyMA/TxBody.hs]: Sorry, I totally messed up. This is the only era that provides a special pattern synonym for deconstructing only.

We need to get rid of it and other accessors (like `inputs'`, `outputs'`, etc.) in other eras, but we  probably should do that in a separate PR. I created an issue for this : #3130 
Could you please revert this change of tick to underscore and the prefix as well `certs'` -> `matbCerts_` for this pattern `MATXBody'`/`MATXBody_`. Alternatively you can just get rid of this pattern synonym in this PR if you prefer.

**Review by lehins — CHANGES_REQUESTED** [review]
Sorry, more review comments. I am truly enjoying seeing this cleanup!

File: eras/alonzo/impl/src/Cardano/Ledger/Alonzo/TxBody.hs (hunk 12 above)
Comment (aniketd) [reply eras/alonzo/impl/src/Cardano/Ledger/Alonzo/TxBody.hs]: Sure! :smile: I did think about it before too, but thought that using it in one place will give enough safety. I will switch it out.

File: eras/babbage/impl/src/Cardano/Ledger/Babbage/TxBody.hs (hunk 13 above)
Comment (aniketd) [reply eras/babbage/impl/src/Cardano/Ledger/Babbage/TxBody.hs:40]: :+1:

File: eras/babbage/impl/src/Cardano/Ledger/Babbage/TxBody.hs (hunk 14 above)
Comment (aniketd) [reply eras/babbage/impl/src/Cardano/Ledger/Babbage/TxBody.hs]: Noted! All `[Era]TxBody` pattern synonyms should need the switch :+1:

File: eras/alonzo/impl/src/Cardano/Ledger/Alonzo/TxBody.hs (hunk 11 above)
Comment (aniketd) [reply eras/alonzo/impl/src/Cardano/Ledger/Alonzo/TxBody.hs:41]: Oops! Missed it.

File: eras/shelley-ma/impl/src/Cardano/Ledger/ShelleyMA/TxBody.hs (hunk 10 above)
Comment (aniketd) [reply eras/shelley-ma/impl/src/Cardano/Ledger/ShelleyMA/TxBody.hs]: I got rid of it!

File: CHANGELOG.md [hunk 15]
```diff
@@ -28,7 +28,107 @@ in the naming of release branches.
...
+    - `_wdrls` to `atbrWdrls`
+    - `_txfee` to `atbrTxFee`
+    - `_vldt` to `atbrValidityInterval`
+    - `_update` to `atbrUpdate`
+    - `_reqSignerHashes` to `atbrReqSignerHashes`
+    - `_mint` to `atbrMint`
+    - `_scriptIntegrityHash` to `atbrScriptIntegrityHash`
+    - `_adHash` to `atbrAuxDataHash`
+    - `_txnetworkid` to `atbrTxNetworkId`
+  - `Alonzo.TxBody.AlonzoTxBodyRaw` pattern synonym
```
Comment (lehins) [inline CHANGELOG.md]:
```suggestion
  - `Alonzo.TxBody.AlonzoTxBody` pattern synonym
```

**Review by lehins — APPROVED** [review]
Looks awesome. Thank you!

**Comment by aniketd** [comment]
Squashed and rebased!

**Comment by aniketd** [comment]
Can I merge this?

**Comment by teodanciu** [comment]
@aniketd It's approved,  good to merge I think! :rocket: 
We can always change later if some details could still be improved

**Comment by JaredCorduan** [comment]
go for it @aniketd !
